src/Agent.Sdk/SecretMasking/RegexSecret.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Agent.Sdk.SecretMasking;
+
+internal sealed class RegexSecret : ISecret
+{
+    public RegexSecret(String pattern, bool compiled = false)
```

Comment:
>compiled [](http://example.com/codeflow?start=44&length=8)

In addition to the `Compiled` option, other settings can have performance impact, e.g., `ExplicitCapture` is helpful to reduce overhead of named groups. Moving forward, the .NET 7/8 `NonBacktracking` option will be something we want to explore. So general note here is that we should probably consider burning RegexOptions here as an argument, allowing users to provide arbitrary combinations of them. In theory, a masker might want a case-insensitive regex, for example.

Reply:
@michaelcfanning, fixed. Thank you!